ta4j-core/src/main/java/org/ta4j/core/num/DoubleNum.java

```diff
@@ -35,7 +35,7 @@
  */
 public class DoubleNum implements Num {
 
-    private static final DoubleNum ZERO = DoubleNum.valueOf(0);
+    public static final DoubleNum ZERO = DoubleNum.valueOf(0);
```

Comment:
why making (only) this one public?

Reply:
I am unsure if it makes sense at all to specify the precision to another value than the default value. Can you imagine a use case that would work e.g. where you could work with a higher or lower precision.

Example of calculation in `SMAIndicator`:
``` java
    @Override
    protected Num calculate(int index) {
        Num sum = zero();                                                                                  // default precision from Num.ZERO
        for (int i = Math.max(0, index - barCount + 1); i <= index; i++) {
            sum = sum.plus(indicator.getValue(i));                                            // precision from Num.ZERO used
        }
        final int realBarCount = Math.min(barCount, index + 1);
        return sum.dividedBy(numOf(realBarCount));                                    // precision from Num.ZERO used as well
    }
    @Override
    public String toString() {
        return getClass().getSimpleName() + " barCount: " + barCount;
    }
```

Reply:
>  because it uses the zero() with the inherited precision 
okay right, now I see how once could configure the precision. 

static DecimalNumBarSeries(..) // uses default precision
static DoubleNumBarSeries(..).` // uses default precision

sounds interesting, but I think so far it is enough to have the BaseBarSeriesBuilder with `DecimalNum` as default and with the possibility to set it to `DoubleNum` if required.

> I can provide a PR (after this was merged) where the user can also set another precision for DoubleNum.

Regarding this I am unsure if there is a demand for this feature. In general I must admit that I have the feeling that the whole `Num` abstraction is not relevant for the vast majority of users.